### PR TITLE
Fixing some minor typos

### PR DESCRIPTION
**Pull request by RachithP**

### CODE REVIEW

**Review by mikeroberts3000 — APPROVED** [review]
Good catch.

**Review by mikeroberts3000 — APPROVED** [review]

**Review by mikeroberts3000 — CHANGES_REQUESTED** [review]
Let's talk about this `<format>` before merging.

File: cpp/unreal_plugins/SpCore/Source/SpCore/EngineActor.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include <UObject/NameTypes.h>  // FName
 #include <UObject/UnrealType.h> // FProperty, FStructProperty
 
+#include "SpCore/Assert.h"
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/EngineActor.cpp:15]: Good catch.

File: cpp/unreal_plugins/SpCore/Source/SpCore/Log.cpp [hunk 2]
```diff
@@ -5,7 +5,6 @@
 #include "SpCore/Log.h"
 
 #include <filesystem>
-#include <format>
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/Log.cpp:8]: I think the issues with `<format>` were linker errors, so it is still safe to include it here.

File: cpp/unreal_plugins/SpCore/Source/SpCore/Log.cpp [hunk 3]
```diff
@@ -15,6 +14,7 @@
 #include <HAL/Platform.h>            // TEXT
 #include <Logging/LogMacros.h>       // DECLARE_LOG_CATEGORY_EXTERN, DEFINE_LOG_CATEGORY, UE_LOG
 
+#include "SpCore/Assert.h"
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/Log.cpp:17]: Good catch.

File: cpp/unreal_plugins/SpCore/Source/SpCore/Log.cpp [hunk 4]
```diff
@@ -40,7 +40,7 @@ void Log::logUnreal(const std::string& str)
 std::string Log::getPrefix(const std::filesystem::path& current_file, int current_line)
 {
     // We don't use Std::toString(current_line) because we want to pad with leading zeros.
-    return "[SPEAR | " + getCurrentFileAbbreviated(current_file) + ":" + std::format("{:04}", current_line) + "] ";
+    return "[SPEAR | " + getCurrentFileAbbreviated(current_file) + ":" + toString("{:04}", current_line) + "] ";
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/Log.cpp:43]:
```
    #include <boost/predef.h> // BOOST_COMP_CLANG, BOOST_COMP_MSVC

    // TODO: remove platform-specific logic
    #if BOOST_COMP_MSVC
        return "[SPEAR | " + getCurrentFileAbbreviated(current_file) + ":" + std::format("{:04}", current_line) + "] ";
    #elif BOOST_COMP_CLANG
        // whatever
    #elif
        #error
    #endif
```

File: cpp/unreal_plugins/SpCore/Source/SpCore/StableNameComponent.cpp [hunk 5]
```diff
@@ -7,6 +7,7 @@
 #include <Components/ActorComponent.h>
 #include <GameFramework/Actor.h>
 
+#include "SpCore/Assert.h"
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/StableNameComponent.cpp:10]: Good catch.

File: cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h [hunk 6]
```diff
@@ -428,13 +428,12 @@ class SPCORE_API Unreal
     {
         auto components = Std::toVector<TActorComponent*>(
             getComponentsByType<TActorComponent>(actor) |
-            std::views::transform([&tags](auto component) { return std::make_pair(component, getComponentHasTags(actor, tags)); }) |
+            std::views::transform([&tags](auto component) { return std::make_pair(component, getComponentHasTags(component, tags)); }) |
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h:431]: Good catch. Also lol how did this code compile? I'm trying to pass an `AActor*` to a function that is expecting a `UActorComponent*`.

File: cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h [hunk 7]
```diff
@@ -563,8 +562,7 @@ class SPCORE_API Unreal
         }
         if (vec.size() == 0) {
             return default_val;
-        }
-        else {
+        } else {
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h:565]: Good catch.

File: cpp/unreal_plugins/SpCore/Source/SpCore/UserInputComponent.cpp [hunk 8]
```diff
@@ -2,8 +2,6 @@
 // Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
 //
 
-#pragma once
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/UserInputComponent.cpp:5]: Good catch.

File: cpp/unreal_plugins/SpCore/Source/SpCore/UserInputComponent.cpp [hunk 9]
```diff
@@ -16,6 +14,7 @@
 #include <GameFramework/PlayerController.h>
 #include <GameFramework/PlayerInput.h>
 
+#include "SpCore/Assert.h"
```
Comment (mikeroberts3000) [inline cpp/unreal_plugins/SpCore/Source/SpCore/UserInputComponent.cpp:17]: Good catch.

File: cpp/unreal_projects/SpearSim/Python/validate_scene_kujiale.py
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Python/validate_scene_kujiale.py:1]: Good.

File: cpp/unreal_projects/SpearSim/Source/SpearSim/SpPlayerController.cpp [hunk 10]
```diff
@@ -4,6 +4,7 @@
 
 #include "SpearSim/SpPlayerController.h"
 
+#include "SpCore/Assert.h"
```
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Source/SpearSim/SpPlayerController.cpp:7]: Good catch.

File: cpp/unreal_projects/SpearSim/Source/SpearSimEditor/DebugWidget.cpp [hunk 11]
```diff
@@ -14,6 +14,7 @@
 #include <Math/Vector.h>
 #include <PhysicsEngine/BodyInstance.h>
 
+#include "SpCore/Assert.h"
```
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Source/SpearSimEditor/DebugWidget.cpp:17]: Good catch.

File: cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h (hunk 6 above)
Comment (RachithP) [reply cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h:431]: It did not compile on Linux, but not sure about Windows.

**Pull request merged**